org.adempiere.base/src/org/compiere/process/SvrProcess.java

```diff
@@ -901,4 +910,33 @@ private List<Field> getFieldsWithParameters(Class<?> clazz) {
 		return Collections.emptyList();
 	}
 
+	/**
+	 * Refer {@link #addDownloadFile(File, boolean)} with onlySuccess = true
+	 * @param file
+	 */
+	public void addDownloadFile(File file) {
+		addDownloadFile(file, true);
+	}
+
+	/**
+	 * For backward compatibility call {@link IProcessUI#download(File)}<br/>
+	 * Stored file path in the {@link ProcessInfoLog#m_P_Msg} with the {@link ProcessInfoLog#getPInstanceLogType} set to {@link X_AD_PInstance_Log#PINSTANCELOGTYPE_FilePath} 
+	 * @param file
+	 * @param onlySuccess normal file download keep on buffer log so appear only success case. In case want to show file ever un-success process then parse false  
+	 */
+	public void addDownloadFile(File file, boolean onlySuccess) {
+		if (processUI != null)
+			processUI.download(file);
+		
+		// store file path to audit log to get back for case infowindow

```

Comment:
this is actually less useful than it seems due to:
- file is usually temporary so the link will become invalid after a while
- in a distribute environment, the link is invalid when you look at it from a different node.

Reply:
@CarlosRuiz-globalqss I store only file paths. I use log types to distinguish between them so isn't big change

@hengsin I believe the concerns about data validity in a distributed environment are mitigated by this data use only for process instance

store file path also to support resolve IDEMPIERE-5997

Reply:
@hieplq , stored file path is not a reliable solution for IDEMPIERE-5997. In a multiple server node environment, you might land on a different server node on your next login and the stored path wouldn't work.

Reply:
I believe storing the file path in the log is acceptable for now. Since the job still runs on the old server, it can retrieve the files and complete its tasks (sending emails and creating attachments) even after finishing execution.

To address your concern about log, I've decided to close this pull request and propose a new solution that doesn't rely on logs. I've moved the downloadFiles from AbstractProcessDialog to ProcessInfo instead